common/src/main/scala/explore/components/AgGridHelpers.scala

```diff
@@ -0,0 +1,258 @@
+   * be chained onto the return value.
+   *
+   * @param lens Lens to go between the Row value and the Cell value.
+   * @param validFormat The ValidFormatInput for the editor.
+   * @param changeAuditor The ChangeAuditor for the editor.
+   * @param disabled Whether editing should be disabled or not.
+   * @return An AgGridColumn Builder
+   */
+  def editableViewColumn[A, B](
+    lens:          Lens[A, B],
+    validFormat:   ValidFormatInput[B] = ValidFormatInput.id,
+    changeAuditor: ChangeAuditor[B] = ChangeAuditor.accept[B],
+    disabled:      Boolean = false
+  )(implicit eq:   Eq[B]): AgGridColumn.ColDef.Builder =
+    AgGridColumn.ColDef
+      .editable(!disabled)
+      .valueGetter(getter(lens))
+      .valueSetter(noopSetter)
+      .cellRendererFramework(renderer(validFormat.reverseGet))
+      .cellEditorFramework(makeFormInputEVEditor(validFormat, changeAuditor))
+      .suppressKeyboardEvent(suppressKeys)
+
+  /**
+   * Create a AgGridColumn Builder for a column which edits the value via a EnumViewSelect.
+   * Obviously, the cell value must have an Enumerated instance.
+   * Since this returns a Builder, additional column configuration methods can
+   * be chained onto the return value.
+   *
+   * @param lens Lens to go between the Row value and the Cell value.
+   * @param disabled Whether editing should be disabled or not.
+   * @param exclude A Set of values to exclude from the dropdown list.
+   * @param includeCurrent Used with `exclude` to always include the currently selected value in the dropdown list, even if it is in the exclude set.
+   * @return An AgGridColumn Builder.
+   */
+  def editableEnumViewColumn[A, B](
+    lens:           Lens[A, B],
+    disabled:       Boolean = false,
+    exclude:        Set[B] = Set.empty[B],
+    includeCurrent: Boolean = false
+  )(implicit enum:  Enumerated[B], display: Display[B]): AgGridColumn.ColDef.Builder =
+    AgGridColumn.ColDef
+      .editable(!disabled)
+      .valueGetter(getter(lens))
+      .valueSetter(noopSetter)
+      .cellRendererFramework(renderer[B](display.shortName))
+      .cellEditorFramework(makeFormInputEnumEditor(exclude, includeCurrent))
+      .suppressKeyboardEvent(suppressKeys)
+
+  /**
+   * Create a column containing a button.
+   * The button is passed in to allow maximum flexibility, but the
+   * onClick handler should not be specified for the button, since
+   * it will be added here.
+   *
+   * The [A] type is the type of the entire data row.
+   *
+   * @param button The button to put in the cell.
+   * @param onClick The onClick handler for the button.
+   * @param disabled Whether the button should be disabled.
+   * @return An AgGridColumn Builder.
+   */
+  def buttonViewColumn[A](
+    @unused button:   Button,
+    @unused onClick:  A => Callback,
+    @unused disabled: Boolean = false
+  ): AgGridColumn.ColDef.Builder = {
+    val component = ScalaComponent
```

Comment:
Can we try doing `button.addModifiers(...).render` here and see if that saves us from creating the wrapper Scala component?

Reply:
Sorry, it's not `render` that we need to call but `component`, which is actually protected in `GenericJsComponentAC`. Maybe we could make it public in react-common...

Reply:
Ok, that wouldn't work either... `Button` encapsulates other properties... Let's leave it as it is.